modules/standard/Filerator.chpl

```diff
@@ -153,33 +153,41 @@ iter walkdirs(path: string=".", topdown=true, depth=max(int), dotfiles=false,
 }
 
 
-/* iter wordexp(pattern="*")
+var chpl_first_walkdirs_sorted_warning = false;
 
-   wordexp() gives a glob-like capability, implemented with C's wordexp()
-   (which is, itself a routine that provides a glob-like capability :)
-
-     * pattern: the glob pattern to match against
-  
-   By default, it will list all files/directories in the current directory
-*/
-
-iter wordexp(pattern="*") {
-  extern type wordexp_t;
-  extern proc wordexp(pattern:c_string, ref ret_glob: wordexp_t, 
-                      flags_c_int): c_int;
-  extern proc chpl_wordexp_num(x:wordexp_t): size_t;
-  extern proc chpl_wordexp_index(x:wordexp_t, idx:size_t): c_string;
-  extern proc wordfree(ref glb:wordexp_t);
-
-  var glb : wordexp_t;
+//
+// Here's a parallel version
+//
+iter walkdirs(path: string=".", topdown=true, depth=max(int), hidden=false, 
+              followlinks=false, sort=false, param tag: iterKind): string 
+       where tag == iterKind.standalone {
+
+  compilerWarning("Parallel invocations of walkdirs() will only parallelize the first-level directories at present");
+  if (sort && chpl_first_walkdirs_sorted_warning) {
+    chpl_first_walkdirs_sorted_warning = true;
+    warning("sorting has no effect for a parallel invocation of walkdirs()");
+  }
 
-  const err = wordexp(pattern:c_string, glb, 0);
+  if (topdown) then
+    yield path;
```

Comment:
I don't see a cute/clever way around this but multiple yields will prevent iterator inlining. I don't think it really matters for these iterators since their io bound anyways.